data_log.py

```diff
@@ -486,6 +527,254 @@ def mean_absolute_percentage_error(y_true, y_pred):
+        sd = train[:,i].std()
+        mean_std_values[column] = (mu, sd)
+        # Standardize training data
+        train[:,i] = (train[:,i] - mu)/sd
+    
+    # Create overlapping windows with training data
+    X_train, y_train = overlap_windows(train, timesteps, steps_ahead)
+    
+    # Testing / Validation period
+    # ---------------------------
+    sv = et 
+    ev = sv + dt.timedelta(days=int((1-train_percent)*window_days)+1)
+    test = data.loc[sv:ev].values
+    
+    # Transform testing/validation data set
+    for i, column in enumerate(columns):
+        # Use mean and standard deviation from the
+        # training data set
+        mu = mean_std_values[column][0]
+        sd = mean_std_values[column][1]
+        # Standardize test data
+        test[:,i] = (test[:,i] - mu)/sd
+    
+    # Create overlaping windows with test data
+    X_test, y_test = overlap_windows(test, timesteps, steps_ahead)
+    
+    return mean_std_values, X_train, y_train, X_test, y_test
+
+
+# In[ ]:
+
+
+# Day-ahead (i.e. 24-hours ahead) short-term load prediction with time-series data
+START_DATE = '2014-01-09'  # starting date of the time-series data
+HISTORY_SIZE = 7*24        # window size in hours for the past history values
+STEPS_AHEAD = 24           # hours-ahead for prediction
+WINDOW_SIZE_DAYS = 300     # window size in days for training and validation
+
+
+# In[ ]:
+
+
+# Training, Validation and Testing data sets
+mean_std_values, X_train, y_train, X_test, y_test = samples_timesteps_features(
+    dataframe, ['Load', 'Temperature'], 
+    start_date=START_DATE, 
+    timesteps=HISTORY_SIZE, 
+    steps_ahead=STEPS_AHEAD,
+    window_days=WINDOW_SIZE_DAYS)
+
+
+# In[ ]:
+
+
+print('Training data shape:')
+print(X_train.shape, y_train.shape)
+print('Validation/Testing data shape:')
+print(X_test.shape, y_test.shape)
+
+
+# In[ ]:
+
+
+BATCH_SIZE = 256       # batch size
+EPOCHS = 10            # number of epochs for training
+STEPS_PER_EPOCH = 100  # steps per epoch for training
+VAL_STEPS = 50         # validation steps during training
+WAIT = 50              # patience for early stopping
+LR = 1e-3              # learning rate
+BUFFER_SIZE = 10000    # buffer size for shuffling batches using tf.data.Dataset
+
+
+# In[ ]:
+
+
+# Convert arrays using tf.data.Dataset
+# Training dataset
+train_data = tf.data.Dataset.from_tensor_slices((X_train, y_train))
+train_data = train_data.cache().shuffle(BUFFER_SIZE).batch(BATCH_SIZE).repeat()
+# Validation dataset
+test_data = tf.data.Dataset.from_tensor_slices((X_test, y_test))
+test_data = test_data.batch(BATCH_SIZE).repeat()
+
+
+# #### Convolutional (LSTM) deep ANN using functional `tf.keras` API
+
+# In[ ]:
+
+
+# Convolutional two-layer deep ANN without regularization
+# tf.keras functional API
+input_layer = keras.layers.Input(shape=(X_train.shape[1:]), batch_size=BATCH_SIZE)
+x = keras.layers.LSTM(units=32, return_sequences=True, activation='relu')(input_layer)
```

Comment:
Size of the layer can be optimized. LSTM layers have regularization options which can be activated.